src/services/todoService.ts

```diff
@@ -0,0 +1,74 @@
+import { json } from "express";
+const NO_ID_MESSAGE = "id isn't in the todo list!";
+const LIST_IS_EMPTY = "todo's list is empty!";
+
+export class TodoService{
+    private todos: ListItem[] = [];
+    
+    //GET
+    read(){
+        return this.todos;
+    }
+
+    //POST
+    add(name: string, date: Date){
+        const item = new ListItem(name, date);
+        this.todos.push(item);
+        return item;
+    }
+
+    //DELETE
+    delete(missionToDelete: string){
+        if(this.todos.length === 0){
+            return LIST_IS_EMPTY
+        }
+        const idx =  this.todos.findIndex(mission => mission.id === missionToDelete);
+        if (idx === -1){
+            return NO_ID_MESSAGE;
+        }
```

Comment:
delete should be "idempotent" call. remember what does that mean?  if no let's discuss